core/src/query.rs

```diff
@@ -30,7 +36,122 @@ pub struct MongoQuery {
     pub query_timeout: Option<u32>,
 }
 
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug, PartialOrd, Ord)]
+pub struct Namespace {
+    pub database: String,
+    pub collection: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Translation {
+    pub target_db: String,
+    pub target_collection: Option<String>,
+    pub pipeline: bson::Bson,
+    pub result_set_schema: Schema,
+    pub select_order: Vec<Vec<String>>,
+}
+
+impl Translation {
+    pub fn from_document(doc: &Document) -> Result<Self> {
+        let as_bson = Bson::Document(doc.clone());
+        let deserializer = bson::Deserializer::new(as_bson);
+        let deserializer = serde_stacker::Deserializer::new(deserializer);
+        Deserialize::deserialize(deserializer).map_err(Error::LibmongosqltranslateDeserialization)
+    }
+}
+
+impl Namespace {
+    pub fn from_bson(bson: Bson) -> Result<BTreeSet<Self>> {
+        let deserializer = bson::Deserializer::new(bson);
+        let deserializer = serde_stacker::Deserializer::new(deserializer);
+        Deserialize::deserialize(deserializer).map_err(Error::LibmongosqltranslateDeserialization)
+    }
+}
+
 impl MongoQuery {
+    fn get_sql_query_namespaces(sql_query: &str, db: &String) -> Result<BTreeSet<Namespace>> {
+        let get_namespaces_command = doc! {
+            "command": "getNamespaces",
+            "options": {
+                "sql": sql_query,
+                "db": db,
+            },
+        };
+
+        let returned_doc = handle_libmongosqltranslate_command(get_namespaces_command)?;
+
+        let namespaces: BTreeSet<Namespace> = Namespace::from_bson(
+            returned_doc
+                .get("namespaces")
+                .ok_or(Error::LibmongosqltranslateDocumentKeyMissing(
+                    "getNamespaces".to_string(),
+                    "namespaces".to_string(),
+                ))?
+                .to_owned(),
+        )?;
+
+        Ok(namespaces)
+    }
+
+    fn translate_sql(
+        sql_query: &str,
+        current_db: &String,
+        namespaces: BTreeSet<Namespace>,
+        client: &MongoConnection,
+        db: &Database,
+    ) -> Result<Translation> {
+        let schema_collection = db.collection::<Document>("__sql_schemas_");
```

Comment:
```suggestion
        let schema_collection = db.collection::<Document>("__sql_schemas");
```
The collection that the schema builder tool creates doesn't have the `_` suffix